src/adapters/index.ts

```diff
@@ -20,27 +21,62 @@ const messageKeyErrorParser = StrictSchema<MessageKeyError>()(
   })
 );
 
-export function parseError(error: unknown): string {
+interface MetamaskUserRejectedRequestError {
+  code: 4001;
+  message: string;
+}
+
+const metamaskUserRejectedRequestError = StrictSchema<MetamaskUserRejectedRequestError>()(
+  z.object({
+    code: z.literal(4001),
+    message: z.string(),
+  })
+);
+
+export function isMetamaskUserRejectedRequestError(
```

Comment:
I think that we can move this to the `types.ts` file (at the end type guards are related to types) and move the code from this file to a new `adapters/errors.ts` file, having all this stuff in the `index.ts` doesn't make too much sense (I think that we've already talked about this but just wanted to remind it).

Reply:
> I think that we can move this to the `types.ts` file

Do you mean moving only the type guard?

I thought it made sense to keep the type guard `isMetamaskUserRejectedRequestError`, next to the type `MetamaskUserRejectedRequestError` and the parser `metamaskUserRejectedRequestError`. We could move the type guard to `types.ts` but being such small pieces, don't you see also a benefit on having them all together?

> move the code from this file to a new `adapters/errors.ts `

Done!

> I think that we've already talked about this

We did when we only had one error-related function here, but now it's true that it deserves its own module.

Reply:
> Do you mean moving only the type guard?

I thought it made sense to keep the type guard isMetamaskUserRejectedRequestError, next to the type MetamaskUserRejectedRequestError and the parser metamaskUserRejectedRequestError. We could move the type guard to types.ts but being such small pieces, don't you see also a benefit in having them all together?

I think that is more semantic to have all the stuff related to type checking like guards and types in one place, and then import them from the correct module. For example, I think that it's better to call `types.isMetamaskUserRejectedRequestError` instead of `adapters.errors.isMetamaskUserRejectedRequestError` from the `global.thunks` which is not what you'd expect without taking a look at the code.

Reply:
Well, I see the module `src/utils/types.ts` more for helper types as `AsyncTask` or `Either`, but ok, it's not very relevant.

Do you agree on moving the 3 related elements (type, parser and, type guard) or do you prefer moving only the type guard?

Reply:
Well, the `isMetamaskUserRejectedRequestError` at the end is a helper that you use across the app (for example in the `global.thunks` and in the `error.adapters`)

Reply:
Ok. Done!